build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BuildPlugin.java

```diff
@@ -59,9 +60,9 @@ public void apply(final Project project) {
         }
 
         project.getPluginManager().apply("elasticsearch.java");
-        project.getPluginManager().apply("elasticsearch.publish");
```

Comment:
we should not apply the publish plugin by default when applying the build plugin:
- we don't want all artifacts being published (we only intent to publish ~35 projects instead of 175 having using the build plugin
- by relying on explicitly applying the publish plugin we have more control over what projects we publish